### PR TITLE
Make Queue.Peek inline

### PR DESCRIPTION
**Pull request by jamesqo**
We can trivially make Queue.Peek inline by moving the throw code into a new method. JIT will not inline this as per dotnet/coreclr#6103.

[Test code, before/after asm](https://gist.github.com/jamesqo/6faf5c7c7a6237f74cbd97d485a3fcd1)

@ianhays

### CODE REVIEW

File: src/System.Collections/src/System/Collections/Generic/Queue.cs [hunk 1]
```diff
@@ -328,6 +330,12 @@ private void MoveNext(ref int index)
             index = (tmp == _array.Length) ? 0 : tmp;
         }
 
+        private void ThrowForEmptyQueue()
```
Comment (stephentoub) [inline src/System.Collections/src/System/Collections/Generic/Queue.cs]: Dequeue can call this as well, right?  If nothing else it'd avoid duplicate usage of InvalidOperation_EmptyQueue.  Not enough of a reason to separate it out into its own function, but as long as it is being separated out, might as well do so.

**Comment by ianhays** [comment]
LGTM.

File: src/System.Collections/src/System/Collections/Generic/Queue.cs (hunk 1 above)
Comment (jamesqo) [inline src/System.Collections/src/System/Collections/Generic/Queue.cs]: Good point, I forgot Dequeue does this as well. Have updated.

**Comment by jamesqo** [comment]
Test Innerloop Windows_NT Debug Build and Test

**Comment by stephentoub** [comment]
Thanks.

**Pull request merged**